File: be/app/api/routes/artikel_route.py
from flask import Blueprint
from app.api.controllers import artikel_controller

artikel_bp = Blueprint('artikel', __name__, url_prefix='/api/artikel')

@artikel_bp.route('', methods=['GET'])
def get_all():
    return artikel_controller.get_all()

@artikel_bp.route('/popular', methods=['GET'])
def get_popular():
    return artikel_controller.get_popular()

@artikel_bp.route('/tags', methods=['GET'])
def get_tags():
    return artikel_controller.get_tags()


@artikel_bp.route('/my-artikel', methods=['GET'])
def my_artikel():
    return artikel_controller.my_artikel()

@artikel_bp.route('/<item_id>', methods=['GET'])
def get_one(item_id):
    return artikel_controller.get_one(item_id)

@artikel_bp.route('', methods=['POST'])
def create():
    return artikel_controller.create()

@artikel_bp.route('/<item_id>', methods=['PUT', 'PATCH'])
def update(item_id):
    return artikel_controller.update(item_id)

@artikel_bp.route('/<item_id>', methods=['DELETE'])
def delete(item_id):
    return artikel_controller.delete(item_id)

# Like
@artikel_bp.route('/<item_id>/like', methods=['POST'])
def toggle_like(item_id):
    return artikel_controller.toggle_like(item_id)

# Komentar
@artikel_bp.route('/<item_id>/komentar', methods=['GET'])
def get_komentar(item_id):
    return artikel_controller.get_komentar(item_id)

@artikel_bp.route('/<item_id>/komentar', methods=['POST'])
def create_komentar(item_id):
    return artikel_controller.create_komentar(item_id)

@artikel_bp.route('/<item_id>/komentar/<komentar_id>', methods=['PUT', 'PATCH'])
def update_komentar(item_id, komentar_id):
    return artikel_controller.update_komentar(item_id, komentar_id)

@artikel_bp.route('/<item_id>/komentar/<komentar_id>', methods=['DELETE'])
def delete_komentar(item_id, komentar_id):
    return artikel_controller.delete_komentar(item_id, komentar_id)

File: fe/src/services/api/routes/artikel.route.js
import api from "../axios";

export const artikelAPI = {
  getAll: (params) => api.get("/artikel", { params }),
  getPopular: () => api.get("/artikel/popular"),
  getTags: () => api.get("/artikel/tags"),
  getById: (id) => api.get(`/artikel/${id}`),
  getMyArtikel: (params) => api.get("/artikel/my-artikel", { params }),
  
  create: (data, fotoFile) => {
    if (fotoFile) {
      const formData = new FormData();
      formData.append("foto_cover", fotoFile);
      
      // Handle tags array specifically
      if (data.tags && Array.isArray(data.tags)) {
        data.tags.forEach(tag => formData.append("tags", tag));
      }
      
      Object.entries(data).forEach(([key, value]) => {
        if (key !== 'tags' && value !== "" && value !== null && value !== undefined) {
          formData.append(key, value);
        }
      });
      return api.post("/artikel", formData, {
        headers: { "Content-Type": "multipart/form-data" },
      });
    }
    return api.post("/artikel", data);
  },

  update: (id, data, fotoFile) => {
    if (fotoFile) {
      const formData = new FormData();
      formData.append("foto_cover", fotoFile);
      
      // Handle tags array specifically
      if (data.tags && Array.isArray(data.tags)) {
        data.tags.forEach(tag => formData.append("tags", tag));
      }
      
      Object.entries(data).forEach(([key, value]) => {
        if (key !== 'tags' && value !== "" && value !== null && value !== undefined) {
          formData.append(key, value);
        }
      });
      return api.put(`/artikel/${id}`, formData, {
        headers: { "Content-Type": "multipart/form-data" },
      });
    }
    return api.put(`/artikel/${id}`, data);
  },

  delete: (id) => api.delete(`/artikel/${id}`),

  // Like operations
  toggleLike: (id) => api.post(`/artikel/${id}/like`),

  // Komentar operations
  getKomentar: (id, params) => api.get(`/artikel/${id}/komentar`, { params }),
  createKomentar: (id, data) => api.post(`/artikel/${id}/komentar`, data),
  updateKomentar: (id, komentarId, data) => api.put(`/artikel/${id}/komentar/${komentarId}`, data),
  deleteKomentar: (id, komentarId) => api.delete(`/artikel/${id}/komentar/${komentarId}`),
};
